src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkitemOrchestrator.cs

```diff
@@ -24,6 +33,17 @@ public interface IWorkitemOrchestrator
         /// <returns>A task that represents the asynchronous orchestration of the adding operation.</returns>
         Task AddWorkitemAsync(DicomDataset dataset, CancellationToken cancellationToken);
 
+
+        /// <summary>
+        /// Asynchronously orchestrate the canceling of a UPS-RS workitem.
+        /// </summary>
+        /// <param name="dataset">The workitem dataset to add.</param>
+        /// <param name="workitemMetadata">The workitem metadata</param>
+        /// <param name="targetProcedureStepState">The target procedure step state</param>
```

Comment:
targetProcedureStepState is decided by the state machine dictionary. I am thinking there is no guarantee that this could be "CANCELED"? May be left as "SCHEDULED" or "IN PROGRESS" (in future, based on additional validations)?

Reply:
Then the method name is not matching, the method name says we are cancelling

Reply:
Done